okhttp/src/main/java/okhttp3/internal/http2/Http2Codec.java

```diff
@@ -135,22 +145,31 @@ public Http2Codec(
 
   /** Returns headers for a name value block containing an HTTP/2 response. */
   public static Response.Builder readHttp2HeadersList(List<Header> headerBlock) throws IOException {
-    String status = null;
-
+    StatusLine statusLine = null;
     Headers.Builder headersBuilder = new Headers.Builder();
     for (int i = 0, size = headerBlock.size(); i < size; i++) {
-      ByteString name = headerBlock.get(i).name;
+      Header header = headerBlock.get(i);
+
+      // If there were multiple header blocks they will be delimited by nulls. Discard existing
```

Comment:
Just to clarify: the scenario you're envisioning here is the server sends back-to-back header blocks, i.e. one with '100 Continue' followed by '200 OK', and that happens prior to the stalled request having time to read the first header block?